catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DumpCommand.java

```diff
@@ -323,27 +324,52 @@ private void exportMetacard(File dumpLocation, Metacard metacard)
         oos.flush();
       }
     } else {
-      BinaryContent binaryContent;
-      if (metacard != null) {
-        for (MetacardTransformer transformer : transformers) {
-          binaryContent = transformer.transform(metacard, new HashMap<>());
-          if (binaryContent != null) {
-            try (FileOutputStream fos =
-                new FileOutputStream(getOutputFile(dumpLocation, metacard))) {
-              fos.write(binaryContent.getByteArray());
-            }
-            break;
+      List<BinaryContent> binaryContents =
+          getTransform()
+              .transform(
+                  Collections.singletonList(metacard), transformerId, Collections.emptyMap());
+
+      if (CollectionUtils.isNotEmpty(binaryContents)) {
+        if (binaryContents.size() == 1) {
+          try (FileOutputStream fos = new FileOutputStream(getOutputFile(dumpLocation, metacard))) {
+
+            fos.write(binaryContents.get(0).getByteArray());
           }
+        } else {
+          createZip(binaryContents, dumpLocation, metacard);
         }
       }
     }
   }
 
-  private File getOutputFile(File dumpLocation, Metacard metacard) throws IOException {
+  private void createZip(List<BinaryContent> binaryContents, File dumpLocation, Metacard metacard)
+      throws CatalogTransformerException, IOException {
+    File zipPath = new File(getOutputFile(dumpLocation, metacard).getAbsolutePath() + ".zip");
+    try {
+      ZipFile zipFile = new ZipFile(zipPath);
+      int index = 1;
+      for (BinaryContent binaryContent : binaryContents) {
+        ZipParameters p = new ZipParameters();
```

Comment:
✏️ recommend naming this zipParameters or something more meaningful

Reply:
renamed